.gitignore

```diff
@@ -1,3 +1,29 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
```

Comment:
This `.gitignore` file is adding a lot of unnecessary ignores. Ignored files are best chosen carefully so as to not accidentally ignore things which might be wanted. It doesn't seem relevant to the conda packaging anyway.

Reply:
OK I've restored the old version and will do that in a separate PR if required.